app/main/helpers/email.py

```diff
@@ -0,0 +1,51 @@
+from flask import url_for, current_app, render_template
+import mandrill
+from itsdangerous import URLSafeTimedSerializer
+
+from .. import main
+
+
+def send_password_email(user_id, email_address):
+    try:
+        mandrill_client = mandrill.Mandrill(main.config['MANDRILL_API_KEY'])
+        url = generate_reset_url(user_id, email_address)
+        body = render_template("emails/forgotten_password_email.html", url=url)
+        message = {'html': body,
+                   'subject': main.config['FORGOT_PASSWORD_EMAIL_SUBJECT'],
+                   'from_email': main.config['FORGOT_PASSWORD_EMAIL_FROM'],
+                   'from_name': main.config['FORGOT_PASSWORD_EMAIL_NAME'],
+                   'to': [{'email': email_address,
+                           'name': 'Recipient Name',
+                           'type': 'to'}],
+                   'important': False,
+                   'track_opens': None,
+                   'track_clicks': None,
+                   'auto_text': True,
+                   'tags': ['password-resets'],
+                   'headers': {'Reply-To': main.config['FORGOT_PASSWORD_EMAIL_FROM']},  # noqa
+                   'recipient_metadata': [
+                       {'rcpt': email_address,
+                        'values': {'user_id': user_id}}]
+        }
+        result = mandrill_client.messages.send(message=message, async=False,
+                                               ip_pool='Main Pool')
+    except mandrill.Error as e:
+        # Mandrill errors are thrown as exceptions
+        current_app.logger.error("A mandrill error occurred: %s", e)
+        return
+    current_app.logger.info("Sent password email: %s", result)
+
+
+def generate_reset_url(user_id, email_address):
+    encoded_string = str(user_id) + " " + email_address
+    ts = URLSafeTimedSerializer(main.config["SECRET_KEY"])
+    token = ts.dumps(encoded_string, salt=main.config["SECRET_SALT"])
```

Comment:
According to http://pythonhosted.org/itsdangerous/#the-salt salt acts more like a namespace than a hash secret. It should make it impossible to reuse the "reset password" code for other purposes (in case the encoded data is the same), so I think we should rename `SECRET_SALT` into e.g. `RESET_PASSWORD_SALT`